### PR TITLE
docs: update direction of joins section

### PR DESCRIPTION
**Pull request by keydunov**
**Check List**
- [ ] Tests has been run in packages where changes made if available
- [ ] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

[For example #12]

**Description of Changes Made (if issue reference is not provided)**

[Description goes here]

### CODE REVIEW

File: docs/content/Schema/Fundamentals/Working-with-Joins.mdx [hunk 1]
```diff
@@ -730,233 +741,103 @@ cubes:
...
-The first case is to calculate the total revenue per customer. Let's name this
-metric `total_revenue`. We also need to be aware of the fact that orders can be
-placed without customer registration (anonymous customers/guest checkouts).
-Because of anonymous customers, we should start the join from the `orders` onto
-the `customers` cube so that we do not lose data from anonymous orders:
+With the given data model, we have two valid analytics use cases that require different join directions.
 
-<CodeTabs>
+The first case is to calculate the total revenue per customer. 
+To do this, we'll use the `total_revenue` measure that is defined on the orders cube.
```
Comment (igorlukanin) [inline docs/content/Schema/Fundamentals/Working-with-Joins.mdx:747]:
```suggestion
To do this, we'll use the `total_revenue` measure that is defined in the `orders` cube. 
```

File: docs/content/Schema/Fundamentals/Working-with-Joins.mdx [hunk 2]
```diff
@@ -730,233 +741,103 @@ cubes:
...
-placed without customer registration (anonymous customers/guest checkouts).
-Because of anonymous customers, we should start the join from the `orders` onto
-the `customers` cube so that we do not lose data from anonymous orders:
+With the given data model, we have two valid analytics use cases that require different join directions.
 
-<CodeTabs>
+The first case is to calculate the total revenue per customer. 
+To do this, we'll use the `total_revenue` measure that is defined on the orders cube. 
+We need to be aware that orders can be placed without customer registration (anonymous customers/guest checkouts). 
+Therefore, we should start the join from the `orders` cube onto the `customers` cube to ensure that we do not lose data from anonymous orders.
```
Comment (igorlukanin) [inline docs/content/Schema/Fundamentals/Working-with-Joins.mdx:749]:
```suggestion
Therefore, we should ensure that the `orders` cube is on the _left_ side of the join so we do not lose data from anonymous orders.
```
I'm not sure that "start the join" is a good way to put it. Also, in the beginning of this particular page, "_left_" semantics is already used (please check it out).

File: docs/content/Schema/Fundamentals/Working-with-Joins.mdx [hunk 3]
```diff
@@ -730,233 +741,103 @@ cubes:
...
-  },
-
-  dimensions: {
-    id: {
-      sql: `id`,
-      type: `number`,
-      primary_key: true,
-      public: true,
-    },
+The second case is to find customers who have not placed any orders.
```
Comment (igorlukanin) [inline docs/content/Schema/Fundamentals/Working-with-Joins.mdx:794]:
```suggestion
__The second case is to find customers who have not placed any orders.__
```

File: docs/content/Schema/Fundamentals/Working-with-Joins.mdx [hunk 4]
```diff
@@ -730,233 +741,103 @@ cubes:
...
 
-The first case is to calculate the total revenue per customer. Let's name this
-metric `total_revenue`. We also need to be aware of the fact that orders can be
-placed without customer registration (anonymous customers/guest checkouts).
-Because of anonymous customers, we should start the join from the `orders` onto
-the `customers` cube so that we do not lose data from anonymous orders:
+With the given data model, we have two valid analytics use cases that require different join directions.
 
-<CodeTabs>
+The first case is to calculate the total revenue per customer.
```
Comment (igorlukanin) [inline docs/content/Schema/Fundamentals/Working-with-Joins.mdx:746]:
```suggestion
__The first case is to calculate the total revenue per customer.__
```

**Review by igorlukanin — APPROVED** [review]
Looks good. Minor nitpicks in comments.

File: docs/content/Schema/Reference/view.mdx [hunk 5]
```diff
@@ -280,6 +280,26 @@ The `public` property is used to manage the visibility of a view. Valid values
...
+```javascript
+view(`orders`, {
+  public: false
+});
+```
+
+</CodeTabs>
+
+
+You can also use `COMPILE_CONTEXT` for dynamic visibility if necessary.
```
Comment (hassankhan) [inline docs/content/Schema/Reference/view.mdx:300]:
```suggestion
You can also use `COMPILE_CONTEXT` for dynamic visibility if necessary, check out our [Controlling access to cubes and views
](https://cube.dev/docs/recipes/controlling-access-to-cubes-and-views) recipe.
```

**Review by hassankhan — APPROVED** [review]
@keydunov Minor nit, looks great otherwise 👍

**Pull request merged**